naptime-testing/src/test/scala/org/coursera/naptime/NestedMacroCourierTests.scala

```diff
@@ -0,0 +1,95 @@
+package org.coursera.naptime
+
+import com.linkedin.data.schema.DataSchema
+import com.linkedin.data.schema.RecordDataSchema
+import org.coursera.naptime.model.Keyed
+import org.coursera.naptime.resources.TopLevelCourierCollectionResource
+import org.coursera.naptime.router2.Router
+import org.junit.Test
+import org.scalatest.junit.AssertionsForJUnit
+
+/**
+ * This test suite uses Courier to exercise advanced use cases for Naptime.
+ */
+object NestedMacroCourierTests {
+
+  class CoursesResource extends TopLevelCourierCollectionResource[String, Course] {
+    override def resourceName: String = "courses"
+
+    override implicit lazy val Fields: Fields[Course] = BaseFields.withRelated("instructors" -> ResourceName("instructors", 1))
+
+    var relatedFun: Option[String => List[String]] = None
```

Comment:
small nit, you use `fallbackRelatedFunction` below but abbreviate `fun` here. i'd recommend using `function` here too to be consistent.

I also don't follow the need for having a relatedFunction helper and a fallback -- it doesn't look like these helpers are overriding anything in a base class, so i'm not positive why there's even a fallback.